### PR TITLE
Run unit tests sequentially.

### LINKED ISSUE #3647: Flaky tests: `TextRenderer`-related tests randomly fail
* .NET Core Version: 
The current master @ 5.0RC1

**Problem description:**

Multiple observed intermittent (yet persistent) failures that appear to be related to `TextRenderer` and device context (?):

```
    System.Windows.Forms.Tests.ButtonTests.Button_Text_SetCachedPreferredSize_Invalidates [FAIL]
      Assert.Equal() Failure
      Expected: {Width=66, Height=24}
      Actual:   {Width=79, Height=24}
      Stack Trace:
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ButtonTests.cs(954,0): at System.Windows.Forms.Tests.ButtonTests.Button_Text_SetCachedPreferredSize_Invalidates()
```
```
    System.Windows.Forms.Tests.TextRendererTests.TextRenderer_MeasureText_InvokeStringFontSizeTextFormatFlags_ReturnsExpected(text: "string", font: [Font: Name=Segoe UI, Size=9, Units=3, GdiCharSet=1, GdiVerticalFont=False], proposedSize: {Width=100, Height=200}, flags: RightToLeft) [FAIL]
      Assert.Equal() Failure
      Expected: {Width=44, Height=16}
      Actual:   {Width=37, Height=15}
      Stack Trace:
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextRendererTests.cs(487,0): at System.Windows.Forms.Tests.TextRendererTests.TextRenderer_MeasureText_InvokeStringFontSizeTextFormatFlags_ReturnsExpected(String text, Font font, Size proposedSize, TextFormatFlags flags)
```
```
    System.Windows.Forms.Tests.TextRendererTests.TextRenderer_MeasureText_InvokeIDeviceContextStringFontSize_ReturnsExpected(text: "string", font: null, proposedSize: {Width=100, Height=200}) [FAIL]
      Assert.Equal() Failure
      Expected: {Width=44, Height=16}
      Actual:   {Width=37, Height=16}
      Stack Trace:
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextRendererTests.cs(515,0): at System.Windows.Forms.Tests.TextRendererTests.TextRenderer_MeasureText_InvokeIDeviceContextStringFontSize_ReturnsExpected(String text, Font font, Size proposedSize)
```
```
    System.Windows.Forms.Tests.ButtonTests.Button_GetPreferredSize_InvokeSystemGrowOnlyWithText_ReturnsExpected(proposedSize: {Width=1, Height=1}) [FAIL]
      Assert.Equal() Failure
      Expected: {Width=75, Height=30}
      Actual:   {Width=75, Height=31}
      Stack Trace:
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ButtonTests.cs(1247,0): at System.Windows.Forms.Tests.ButtonTests.Button_GetPreferredSize_InvokeSystemGrowOnlyWithText_ReturnsExpected(Size proposedSize)
```

The current hypothesis is that DC may somehow be dirtied that causes tests to fail.
For example, builds for https://github.com/dotnet/winforms/pull/3417 have [failed multiple times](https://dev.azure.com/dnceng/public/_build/results?buildId=746070&view=results) (all in Release), but once I have disabled `ScreenDcCache` in https://github.com/dotnet/winforms/commit/e1539274efdc8c16b90093e322ce81d2709acc02 builds succeeded.

### LINKED ISSUE #4212: Add control rendering tests with enabled visual styles
We have a number of rendering tests, however all of them have been developed for disabled visual styles.
New tests need to be added to:
* [ ] NumericUpDownTests (https://github.com/dotnet/winforms/blob/master/src/System.Windows.Forms/tests/UnitTests/NumericUpDownTests.cs)
* [ ] ButtonRenderingTests (https://github.com/dotnet/winforms/blob/master/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ButtonRenderingTests.cs)

Relates to https://github.com/dotnet/winforms/pull/4197

### LINKED ISSUE #6610: ToolStrip tests is flaky in arm64 and x86 CI build
* .NET Core Version: 7.0

* Have you experienced this same bug with .NET Framework?: No

**Problem description:**
- `ToolStrip_WndProc_InvokeMouseActivateWithHandle_Success` unit test fails sometime in arm64 CI build for successful PRs, that blocking deployment of fixes into sdk. *Stack trace* ([System.Windows.Forms.Tests_net7.0_arm64.log](https://dev.azure.com/dnceng/public/_build/results?buildId=1584807&view=artifacts&pathAsName=false&type=publishedArtifacts)):
```
System.Windows.Forms.Tests.ToolStripTests.ToolStrip_WndProc_InvokeMouseActivateWithHandle_Success [FAIL]
Assert.Equal() Failure
      Expected: 0
      Actual:   2
      Stack Trace:
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.cs(7139,0): at System.Windows.Forms.Tests.ToolStripTests.ToolStrip_WndProc_InvokeMouseActivateWithHandle_Success()
```
- `ToolStrip_WndProc_InvokeMouseActivate_Success`:
image

**Expected behavior:**
All correct tests pass every time.

**Minimal repro:**
- Run unit tests on arm64 environment (or create a PR with changes, CI build will do it)

### LINKED ISSUE #6729: `Clipboard_SetDataObject_InvokeObjectBoolNotIComDataObject_GetReturnsExpected` is flaky in CI build
* .NET Core Version: 7.0

* Have you experienced this same bug with .NET Framework?: No

**Problem description:**
- Test fails sometimes in CI build for successful PRs, that blocking deployment of fixes into sdk:
`Clipboard_SetDataObject_InvokeObjectBoolNotIComDataObject_GetReturnsExpected`:
Stack trace ([comment](https://github.com/dotnet/winforms/pull/6600#issuecomment-1042437384)):
```
Assert.Equal() Failure\r\nExpected: 1\r\nActual:   MemoryStream { CanRead = True, CanSeek = True, CanTimeout = False, CanWrite = True, Capacity = 70, ... }


Stack trace
   at System.Windows.Forms.Tests.ClipboardTests.Clipboard_SetDataObject_InvokeObjectBoolNotIComDataObject_GetReturnsExpected(Object data, Boolean copy) in /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ClipboardTests.cs:line 279
```

**Expected behavior:**
- All correct tests pass every time.

**Minimal repro:**
- Run CI build

### LINKED ISSUE #8601: [flaky test] embeddedresourcetests.embeddedresource_resourcesexist_cursor
Runfo Tracking Issue: [[flaky test] embeddedresourcetests.embeddedresource_resourcesexist_cursor](https://runfo.azurewebsites.net/tracking/issue/457)
|Build|Definition|Kind|Run Name|
|---|---|---|---|

Build Result Summary
|Day Hit Count|Week Hit Count|Month Hit Count|
|---|---|---|
|0|0|0|

### LINKED ISSUE #8602: [flaky test] com2interop.com2fontconvertertests.com2fontconverter_convertmanagedtonative
Runfo Tracking Issue: [[flaky test] com2interop.com2fontconvertertests.com2fontconverter_convertmanagedtonative](https://runfo.azurewebsites.net/tracking/issue/458)
|Build|Definition|Kind|Run Name|
|---|---|---|---|
|[168274](https://dev.azure.com/dnceng-public/public/_build/results?buildId=168274)|[dotnet-winforms CI](https://dnceng-public.visualstudio.com/public/_build?definitionId=76)|[PR 8620](https://github.com/dotnet/winforms/pull/8620)|Windows_arm64-xunit|
|[165414](https://dev.azure.com/dnceng-public/public/_build/results?buildId=165414)|[dotnet-winforms CI](https://dnceng-public.visualstudio.com/public/_build?definitionId=76)|[PR 8592](https://github.com/dotnet/winforms/pull/8592)|Windows_x64-xunit|
|[164111](https://dev.azure.com/dnceng-public/public/_build/results?buildId=164111)|[dotnet-winforms CI](https://dnceng-public.visualstudio.com/public/_build?definitionId=76)|[PR 8588](https://github.com/dotnet/winforms/pull/8588)|Windows_x86-xunit|
|[164100](https://dev.azure.com/dnceng-public/public/_build/results?buildId=164100)|[dotnet-winforms CI](https://dnceng-public.visualstudio.com/public/_build?definitionId=76)|[PR 8574](https://github.com/dotnet/winforms/pull/8574)|Windows_arm64-xunit|

Build Result Summary
|Day Hit Count|Week Hit Count|Month Hit Count|
|---|---|---|
|0|0|4|

### LINKED ISSUE #8609: [flaky test] com2interop.com2pictureconvertertests.convertmanagedtonative_bitmap
Runfo Tracking Issue: [[flaky test] com2interop.com2pictureconvertertests.convertmanagedtonative_bitmap](https://runfo.azurewebsites.net/tracking/issue/462)
|Build|Definition|Kind|Run Name|
|---|---|---|---|

Build Result Summary
|Day Hit Count|Week Hit Count|Month Hit Count|
|---|---|---|
|0|0|0|

### LINKED ISSUE #8611: [Flaky Test] menustriptests.menustrip_wndproc_invokemouseactivate_success
Runfo Tracking Issue: [[Flaky Test] menustriptests.menustrip_wndproc_invokemouseactivate_success](https://runfo.azurewebsites.net/tracking/issue/463)
|Build|Definition|Kind|Run Name|
|---|---|---|---|

Build Result Summary
|Day Hit Count|Week Hit Count|Month Hit Count|
|---|---|---|
|0|0|0|

### LINKED ISSUE #8632: [Flaky Test] com2fontconvertertests.com2fontconverter_convertnativetomanaged
Runfo Tracking Issue: [[Flaky Test] com2fontconvertertests.com2fontconverter_convertnativetomanaged](https://runfo.azurewebsites.net/tracking/issue/465)
|Build|Definition|Kind|Run Name|
|---|---|---|---|

Build Result Summary
|Day Hit Count|Week Hit Count|Month Hit Count|
|---|---|---|
|0|0|0|

### PR DESCRIPTION
**Pull request by dreddy-work**
Replacing #8772.

changes:
- Serialize execution of unit tests on build agents.
- Block tests against `hsplit.cur` and `vsplit.cur`. They need special attention with respect to disposing them. Possibly a bug that has been there since Framework.
- Remove collection definition that is not necessary anymore.
- Add retry count (1).

Tracking issue to regroup unit tests as well as fixing UI tests are tracked by #8810.

Fixes #6610 Fixes #4212 fixes #6729 fixes #8632 fixes #8602 fixes #8609 fixes #8601 fixes #8611 fixes #3647 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8864)

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -28,6 +28,11 @@
     <EnableTrimAnalyzer Condition="$(IsTestProject) or $(IsAnalyzerProject) or $(MSBuildProjectDirectory.Contains('\tests\'))">false</EnableTrimAnalyzer>
   </PropertyGroup>
 
+  <!-- Making all tests run sequentially untill we regroup tests. Tracked under issue https://github.com/dotnet/winforms/issues/8810. -->
```
Comment (JeremyKuhne) [inline Directory.Build.props:31]:
```suggestion
  <!-- Making all tests run sequentially until we regroup tests. Tracked under issue https://github.com/dotnet/winforms/issues/8810. -->
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CursorsTests.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public static IEnumerable<object[]> Cursors_TestData()
             yield return new object[] { I(() => Cursors.Default) };
             yield return new object[] { I(() => Cursors.Hand) };
             yield return new object[] { I(() => Cursors.Help) };
-            yield return new object[] { I(() => Cursors.HSplit) };
+            // yield return new object[] { I(() => Cursors.HSplit) };
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CursorsTests.cs:25]: Why are these commented out?

**Review by JeremyKuhne — CHANGES_REQUESTED** [review]
A few minor things

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CursorsTests.cs (hunk 2 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CursorsTests.cs:25]: Highlighted in the tracking issue. These Cursor are currently loaded as if they are owned by the process and disposed when done. Resulting in exception when reused.

**Review by JeremyKuhne — APPROVED** [review]

**Pull request merged**